### PR TITLE
Add all providers

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

**Comment by Kilo-Loco** [comment]
Are providers already determined internally or is there a reference PR to check this against?

File: src/routes/docs/products/messaging/providers/apns/+page.markdoc [hunk 1]
```diff
@@ -47,6 +47,45 @@ After adding the following details, click **Save and continue** to enable the pr
...
+{% tabsitem #console title="Console" %}
+To send a test message, navigate to **Messaging** > **Messages** > {% icon icon="plus" size="m" /%} **Create message** > **Push notifcation**.
+[SCREENSHOT]
+
+Add your message and in the targets step, select one of your test targets. Set the schedule to **Now** and click **Send**.
+
+Verify that you can receive the message in your inbox. If not
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/providers/apns/+page.markdoc:62]: @abnegate @stnguyen90 Code example for sending a message through this provider

File: src/routes/docs/products/messaging/providers/fcm/+page.markdoc [hunk 2]
```diff
@@ -1,14 +1,90 @@
...
+
+Add your message and in the targets step, select one of your test targets. Set the schedule to **Now** and click **Send**.
+
+Verify that you can receive the message in your inbox. If not
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+To send a message programmatically, use an [Appwrite Server SDK](/docs/sdks#server).
+{% multicode %}
+```js
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/providers/fcm/+page.markdoc:61]: @abnegate @stnguyen90 Code example for sending a message through this provider

File: src/routes/docs/products/messaging/providers/mailgun/+page.markdoc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Add your message and in the targets step, select one of your test targets. Set the schedule to **Now** and click **Send**.
+
+Verify that you can receive the message in your inbox. If not
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+To send a message programmatically, use an [Appwrite Server SDK](/docs/sdks#server).
+{% multicode %}
+```js
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/providers/mailgun/+page.markdoc:74]: @abnegate @stnguyen90 Code example for sending a message through this provider

File: src/routes/docs/products/messaging/providers/msg91/+page.markdoc [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+Add your message and in the targets step, select one of your test targets. Set the schedule to **Now** and click **Send**.
+
+Verify that you can receive the message in your inbox. If not
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+To send a message programmatically, use an [Appwrite Server SDK](/docs/sdks#server).
+{% multicode %}
+```js
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/providers/msg91/+page.markdoc:61]: @abnegate @stnguyen90 Code example for sending a message through this provider

File: src/routes/docs/products/messaging/providers/sendgrid/+page.markdoc [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+Add your message and in the targets step, select one of your test targets. Set the schedule to **Now** and click **Send**.
+
+Verify that you can receive the message in your inbox. If not
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+To send a message programmatically, use an [Appwrite Server SDK](/docs/sdks#server).
+{% multicode %}
+```js
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/providers/sendgrid/+page.markdoc:74]: @abnegate @stnguyen90 Code example for sending a message through this provider

File: src/routes/docs/products/messaging/providers/telesign/+page.markdoc [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+Add your message and in the targets step, select one of your test targets. Set the schedule to **Now** and click **Send**.
+
+Verify that you can receive the message in your inbox. If not
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+To send a message programmatically, use an [Appwrite Server SDK](/docs/sdks#server).
+{% multicode %}
+```js
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/providers/telesign/+page.markdoc:61]: @abnegate @stnguyen90 Code example for sending a message through this provider

File: src/routes/docs/products/messaging/providers/textmagic/+page.markdoc [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+Add your message and in the targets step, select one of your test targets. Set the schedule to **Now** and click **Send**.
+
+Verify that you can receive the message in your inbox. If not
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+To send a message programmatically, use an [Appwrite Server SDK](/docs/sdks#server).
+{% multicode %}
+```js
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/providers/textmagic/+page.markdoc:61]: @abnegate @stnguyen90 Code example for sending a message through this provider

File: src/routes/docs/products/messaging/providers/twilio/+page.markdoc [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+Add your message and in the targets step, select one of your test targets. Set the schedule to **Now** and click **Send**.
+
+Verify that you can receive the message in your inbox. If not
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+To send a message programmatically, use an [Appwrite Server SDK](/docs/sdks#server).
+{% multicode %}
+```js
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/providers/twilio/+page.markdoc:61]: @abnegate @stnguyen90 Code example for sending a message through this provider

File: src/routes/docs/products/messaging/providers/vonage/+page.markdoc [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+Add your message and in the targets step, select one of your test targets. Set the schedule to **Now** and click **Send**.
+
+Verify that you can receive the message in your inbox. If not
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+To send a message programmatically, use an [Appwrite Server SDK](/docs/sdks#server).
+{% multicode %}
+```js
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/providers/vonage/+page.markdoc:62]: @abnegate @stnguyen90 Code example for sending a message through this provider

**Review by gewenyu99 — COMMENTED** [review]
@stnguyen90 @abnegate 

Broke off the providers section of the docs here for easier request for code samples + review later <3

**Pull request merged**